docs/source/common/optimizers.rst

```diff
@@ -21,46 +21,117 @@ Manual optimization
+        opt.step()
 
-        # 1. ignore optimizer_idx
-        # 2. `use_pl_optimizer=True` means `opt_g` and `opt_d` will be of type `LightningOptimizer`
-        # `LightningOptimizer` simply wrapped your optimizer and behave the same way !
-        # When calling `optimizer.step`, `LightningOptimizer` will just handle TPU, AMP, accumulate_grad_batches, etc ... for you.
+        # zero grad
+        if batch_idx % 2 == 0:
+            opt.zero_grad()
 
-        # access your optimizers with `use_pl_optimizer=False` or `optimizer.optimizer` when using use_pl_optimizer=True
-        # use_pl_optimizer=True is the default
-        (opt_g, opt_d) = self.optimizers(use_pl_optimizer=True)
 
-        # do anything you want
-        loss_a = ...
+.. tip:: It is good pratice to provide to the optimizer a ``closure function`` that perform a ``forward and backward`` evaluation of your model. It is optional for most optimizers, but make your code compatible if you switch to an optimizer which requires a closure.
 
-        # use self.backward which will also handle scaling the loss when using amp
-        self.manual_backward(loss_a, opt_g)
-        opt_g.step()
+Here is the same example as above using a ``closure``.
 
+.. code-block:: python
+
+    def training_step(batch, batch_idx, optimizer_idx):
+        opt = self.optimizers()
+
+        def forward_and_backward():
+            loss = self.compute_loss(batch)
+            self.manual_backward(loss)
 
-        # do anything you want
-        loss_b = ...
+        opt.step(closure=forward_and_backward)
 
-        # pass in any args that loss.backward() normally takes
-        self.manual_backward(loss_b, opt_d, retain_graph=True)
-        self.manual_backward(loss_b, opt_d)
-        opt_d.step()
+        # zero grad
+        if batch_idx % 2 == 0:
+            opt.zero_grad()
+
+
+.. code-block:: python
 
-        # log losses
-        self.log('loss_a', loss_a)
-        self.log('loss_b', loss_b)
+    # Scenario for a GAN.
 
-.. note:: This is only recommended for experts who need ultimate flexibility
+    def training_step(...):
+        opt_gen, opt_dis = self.optimizers()
 
-Manual optimization does not yet support accumulated gradients but will be live in 1.1.0
+        # compute generator loss
+        loss_gen = self.compute_generator_loss(...)
+
+        # zero_grad needs to be called before backward
+        opt_gen.zero_grad()
+        self.manual_backward(loss_gen)
+        opt_gen.step()
+
+        # compute discriminator loss
+        loss_dis = self.compute_discriminator_loss(...)
+
+        # zero_grad needs to be called before backward
+        opt_dis.zero_grad()
+        self.manual_backward(loss_dis)
+        opt_dis.step()
+
+
+.. note:: ``LightningOptimizer`` provides a ``toggle_model`` function as a ``context manager`` for advanced users. It can be useful when performing ``gradient accumulation with several optimizers`` and training with ``multiple gpus``.
```

Comment:
We should reserve backticks for code

```suggestion
.. note:: ``LightningOptimizer`` provides a ``toggle_model`` function as a ``@context_manager`` for advanced users. It can be useful when performing gradient accumulation with several optimizers or training in a distributed setting.
```

Reply:
Done !